pkgs/hooks/lib/src/config.dart

```diff
@@ -376,11 +386,16 @@ final class LinkInputBuilder extends HookInputBuilder {
   void setupLink({
     required List<EncodedAsset> assets,
     required Uri? recordedUsesFile,
+    required List<EncodedAsset> assetsFromLinking,
   }) {
     _syntax.setup(
       assets: [
         for (final asset in assets) AssetSyntax.fromJson(asset.toJson()),
       ],
+      assetsFromLinking: [
+        for (final asset in assetsFromLinking)
+          AssetSyntax.fromJson(asset.toJson()),
```

Comment:
I see that we are doing this above as well, but it seems strange that we are converting something to json only to decode it from json again...